pkg/mesos/client.go

```diff
@@ -0,0 +1,90 @@
+package mesos
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+	"github.com/sirupsen/logrus"
+)
+
+// Client is a Mesos client for DC/OS.
+type Client struct {
+	http   *httpclient.Client
+	logger *logrus.Logger
+}
+
+// NewClient creates a new Mesos client.
+func NewClient(baseClient *httpclient.Client, logger *logrus.Logger) *Client {
+	return &Client{
+		http:   baseClient,
+		logger: logger,
+	}
+}
+
+// Masters is returning the mesos masters of the connected cluster.
+func (c *Client) Masters() ([]Master, error) {
+	resp, err := c.http.Get("/mesos_dns/v1/hosts/master.mesos")
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	switch resp.StatusCode {
+	case 200:
+		var hosts []Master
+		err = json.NewDecoder(resp.Body).Decode(&hosts)
+		return hosts, err
+	default:
+		return nil, fmt.Errorf("HTTP %d error", resp.StatusCode)
+	}
+}
+
+// State returns the current State of the mesos master.
+func (c *Client) State() (*State, error) {
+	resp, err := c.http.Get("/mesos/master/state")
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case 200:
+		var state State
+		err = json.NewDecoder(resp.Body).Decode(&state)
+		return &state, err
+	case 503:
```

Comment:
Is `503` really a special case? I think checking for just `200` should be enough (which is also what you're doing in `Masters()`.